ndspy/bmg.py

```diff
@@ -35,8 +40,19 @@ class BMG:
     """
     A class representing a BMG file.
     """
+    messages: list[Message]
+    instructions: list[bytes]
+    labels: list[tuple[int, int]]
+    scripts: list[tuple[int, int]]
 
-    def __init__(self, data=None, *, id=0):
+    id: int
+    encoding: str
```

Comment:
Hm, can we make this a `Literal` too? Valid values for it are listed in the `_ENCODINGS` constant (ignore the `None`, that's just a placeholder because encoding values start at 1)

Reply:
Done. Sadly it's not possible to pass a list of strings to `typing.Literal`, so the string literals have to be duplicated between the `Literal` annotation and the `_ENCODINGS` list. I believe we could do it the other way around, i.e. something like this
```python
import typing

Encoding = typing.Literal['cp1252', 'utf-16', 'shift-jis', 'utf-8']
_ENCODINGS = typing.get_args(Encoding)
```
But that would break 3.7 compatibility. I don't think it's that big of a deal to have them duplicated, but let me know if you'd prefer it be done a different way. (Plus, if `mypy`/some other type checker is eventually introduced as a CI check, it would catch any accidental drift between the type annotation and the `_ENCODINGS` list)

Reply:
I think it's good enough to just add a couple comments to indicate that the two should be kept in sync. I can do that after merging the PR.